src/gc/objecthandle.cpp

```diff
@@ -792,9 +797,18 @@ bool Ref_InitializeHandleTableBucket(HandleTableBucket* bucket, void* context)
         // No free slot.
         // Let's create a new node
         NewHolder<HandleTableMap> newMap;
-        newMap = new HandleTableMap;
+        newMap = new (nothrow) HandleTableMap;
+        if (!newMap)
+        {
+            return false;
+        }
+
+        newMap->pBuckets = new (nothrow) HandleTableBucket * [ INITIAL_HANDLE_TABLE_ARRAY_SIZE ];
+        if (!newMap->pBuckets)
+        {
+            return false;
```

Comment:
>   return false; [](start = 9, length = 16)

This is not exactly a new problem but if it failed to allocate memory you are not cleaning up the memory that was allocated above. It had the same problem exactly when it throws OOM often the program just terminates.